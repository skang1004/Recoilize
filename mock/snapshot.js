export const filteredCurSnapMock = {
  dummyAtom1: {
    contents: {hello: [], hi: []},
    nodeDeps: [],
    nodeToNodeSubscriptions: [],
    type: 'RecoilState',
  },
  listState: {
    contents: [{text: 'list item'}, {text: 'list item'}, {text: 'list item'}],
    nodeDeps: [],
    nodeToNodeSubscriptions: ['selectorTest', 'stateLengths'],
    type: 'RecoilState',
  },
  listState2: {
    contents: [{text: 'list item'}, {text: 'list item'}, {text: 'list item'}],
    nodeDeps: [],
    nodeToNodeSubscriptions: ['stateLengths'],
    type: 'RecoilState',
  },
  selectorTest: {
    contents: 'test',
    nodeDeps: ['listState'],
    nodeToNodeSubscriptions: [],
    type: 'RecoilValueReadOnly',
  },
  stateLengths: {
    contents: 6,
    nodeDeps: ['listState', 'listState2'],
    nodeToNodeSubscriptions: [],
    type: 'RecoilValueReadOnly',
  },
};

export const filteredPrevSnapMock = {
  dummyAtom1: {
    contents: {hello: [], hi: []},
    nodeDeps: [],
    nodeToNodeSubscriptions: [],
    type: 'RecoilState',
  },
  listState: {
    contents: [{text: 'list item'}, {text: 'list item'}, {text: 'list item'}],
    nodeDeps: [],
    nodeToNodeSubscriptions: ['selectorTest', 'stateLengths'],
    type: 'RecoilState',
  },
  listState2: {
    contents: [
      {text: 'list item'},
      {text: 'list item'},
      {text: 'list item'},
      {text: 'list item'},
    ],
    nodeDeps: [],
    nodeToNodeSubscriptions: ['stateLengths'],
    type: 'RecoilState',
  },
  selectorTest: {
    contents: 'test',
    nodeDeps: ['listState'],
    nodeToNodeSubscriptions: [],
    type: 'RecoilValueReadOnly',
  },
  stateLengths: {
    contents: 6,
    nodeDeps: ['listState', 'listState2'],
    nodeToNodeSubscriptions: [],
    type: 'RecoilValueReadOnly',
  },
};

export const componentAtomTreeMock = {
  children: [
    {
      children: [],
      name: '',
      tag: 0,
    },
    {
      children: [
        {
          children: [],
          name: '',
          tag: 0,
        },
        {
          children: [
            {
              children: [],
              name: '',
              tag: 0,
            },
            {
              children: [
                {
                  children: [],
                  name: '',
                  tag: 0,
                },
              ],
              name: '',
              tag: 0,
            },
          ],
          name: '',
          tag: 0,
        },
      ],
      name: '',
      tag: 0,
    },
    {
      children: [
        {
          children: [],
          name: '',
          tag: 0,
        },
        {
          children: [],
          name: '',
          tag: 0,
        },
      ],
      name: '',
      tag: 0,
    },
  ],
  name: '',
  tag: 0,
};
