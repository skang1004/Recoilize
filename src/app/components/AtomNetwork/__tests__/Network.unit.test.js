import React from 'react';
import Network from '../AtomNetwork';
import {render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import '@babel/polyfill';
import {filteredCurSnapMock} from '../../../../../mock/snapshot.js';

afterEach(cleanup);

it('renders & matches snapshot', () => {
  const {asFragment} = render(<Network />);
  expect(asFragment()).toMatchSnapshot();
});

it('loads and displays Network component', () => {
  const {getByTestId} = render(<Network />);
  expect(getByTestId('networkCanvas')).toBeTruthy();
});

it('if empty object props is passed into Network', () => {
  const {asFragment} = render(<Network filteredCurSnap={{}} />);
  expect(asFragment()).toMatchSnapshot();
});

it('if mock data object prop is passed into Network', () => {
  const {asFragment} = render(
    <Network filteredCurSnap={filteredCurSnapMock} />,
  );
  expect(asFragment()).toMatchSnapshot();
});
