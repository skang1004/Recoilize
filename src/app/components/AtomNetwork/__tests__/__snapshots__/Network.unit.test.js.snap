// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if empty object props is passed into Network 1`] = `
<DocumentFragment>
  <div
    class="networkContainer"
  >
    <div
      class="Network"
    >
      <svg
        data-testid="networkCanvas"
        id="networkCanvas"
      >
        <g
          transform="translate(0, 0), scale(1)"
        >
          <defs>
            <marker
              id="arrowhead"
              markerHeight="7"
              markerWidth="7"
              orient="auto"
              refX="13"
              refY="0"
              viewBox="-0 -5 10 10"
              xoverflow="visible"
            >
              <path
                d="M 0,-5 L 10 ,0 L 0,5"
                fill="#474747"
                style="stroke: none;"
              />
            </marker>
          </defs>
        </g>
      </svg>
    </div>
    <input
      id="networkSearch"
      placeholder="search for atoms..."
      type="text"
      value=""
    />
    <div
      class="AtomNetworkLegend"
    >
      <div
        class="AtomLegend"
      />
      <p>
        ATOM
      </p>
      <div
        class="SelectorLegend"
      />
      <p>
        SELECTOR
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`if mock data object prop is passed into Network 1`] = `
<DocumentFragment>
  <div
    class="networkContainer"
  >
    <div
      class="Network"
    >
      <svg
        data-testid="networkCanvas"
        id="networkCanvas"
      >
        <g
          transform="translate(0, 0), scale(1)"
        >
          <defs>
            <marker
              id="arrowhead"
              markerHeight="7"
              markerWidth="7"
              orient="auto"
              refX="13"
              refY="0"
              viewBox="-0 -5 10 10"
              xoverflow="visible"
            >
              <path
                d="M 0,-5 L 10 ,0 L 0,5"
                fill="#474747"
                style="stroke: none;"
              />
            </marker>
          </defs>
          <line
            class="link"
            marker-end="url(#arrowhead)"
          >
            <title />
          </line>
          <line
            class="link"
            marker-end="url(#arrowhead)"
          >
            <title />
          </line>
          <line
            class="link"
            marker-end="url(#arrowhead)"
          >
            <title />
          </line>
          <path
            class="edgepath"
            fill-opacity="1.0"
            id="edgepath0"
            stroke-opacity="1.0"
            style="pointer-events: none; fill: #474747; stroke: #474747; stroke-width: 1px;"
          />
          <path
            class="edgepath"
            fill-opacity="1.0"
            id="edgepath1"
            stroke-opacity="1.0"
            style="pointer-events: none; fill: #474747; stroke: #474747; stroke-width: 1px;"
          />
          <path
            class="edgepath"
            fill-opacity="1.0"
            id="edgepath2"
            stroke-opacity="1.0"
            style="pointer-events: none; fill: #474747; stroke: #474747; stroke-width: 1px;"
          />
          <text
            class="edgelabel"
            font-size="10"
            id="edgelabel0"
            style="pointer-events: none;"
          >
            <textpath
              href="#edgepath0"
              startOffset="50%"
              style="text-anchor: middle; pointer-events: none;"
            />
          </text>
          <text
            class="edgelabel"
            font-size="10"
            id="edgelabel1"
            style="pointer-events: none;"
          >
            <textpath
              href="#edgepath1"
              startOffset="50%"
              style="text-anchor: middle; pointer-events: none;"
            />
          </text>
          <text
            class="edgelabel"
            font-size="10"
            id="edgelabel2"
            style="pointer-events: none;"
          >
            <textpath
              href="#edgepath2"
              startOffset="50%"
              style="text-anchor: middle; pointer-events: none;"
            />
          </text>
          <g
            class="node"
          >
            <circle
              r="5"
              style="fill: #9580FF;"
            />
            <title>
              0
            </title>
            <text
              dy="-3"
              fill="#646464"
              stroke="none"
            >
              dummyAtom1
            </text>
          </g>
          <g
            class="node"
          >
            <circle
              r="5"
              style="fill: #9580FF;"
            />
            <title>
              1
            </title>
            <text
              dy="-3"
              fill="#646464"
              stroke="none"
            >
              listState
            </text>
          </g>
          <g
            class="node"
          >
            <circle
              r="5"
              style="fill: #9580FF;"
            />
            <title>
              2
            </title>
            <text
              dy="-3"
              fill="#646464"
              stroke="none"
            >
              listState2
            </text>
          </g>
          <g
            class="node"
          >
            <circle
              r="5"
              style="fill: #FF80BF;"
            />
            <title>
              3
            </title>
            <text
              dy="-3"
              fill="#646464"
              stroke="none"
            >
              selectorTest
            </text>
          </g>
          <g
            class="node"
          >
            <circle
              r="5"
              style="fill: #FF80BF;"
            />
            <title>
              4
            </title>
            <text
              dy="-3"
              fill="#646464"
              stroke="none"
            >
              stateLengths
            </text>
          </g>
        </g>
      </svg>
    </div>
    <input
      id="networkSearch"
      placeholder="search for atoms..."
      type="text"
      value=""
    />
    <div
      class="AtomNetworkLegend"
    >
      <div
        class="AtomLegend"
      />
      <p>
        ATOM
      </p>
      <div
        class="SelectorLegend"
      />
      <p>
        SELECTOR
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders & matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="networkContainer"
  >
    <div
      class="Network"
    >
      <svg
        data-testid="networkCanvas"
        id="networkCanvas"
      >
        <g
          transform="translate(0, 0), scale(1)"
        >
          <defs>
            <marker
              id="arrowhead"
              markerHeight="7"
              markerWidth="7"
              orient="auto"
              refX="13"
              refY="0"
              viewBox="-0 -5 10 10"
              xoverflow="visible"
            >
              <path
                d="M 0,-5 L 10 ,0 L 0,5"
                fill="#474747"
                style="stroke: none;"
              />
            </marker>
          </defs>
        </g>
      </svg>
    </div>
    <input
      id="networkSearch"
      placeholder="search for atoms..."
      type="text"
      value=""
    />
    <div
      class="AtomNetworkLegend"
    >
      <div
        class="AtomLegend"
      />
      <p>
        ATOM
      </p>
      <div
        class="SelectorLegend"
      />
      <p>
        SELECTOR
      </p>
    </div>
  </div>
</DocumentFragment>
`;
